MekHQ/src/mekhq/gui/adapter/PersonnelTableMouseAdapter.java

```diff
@@ -137,20 +137,40 @@
     private static final String TRUE = String.valueOf(true);
     private static final String FALSE = String.valueOf(false);
 
-    private CampaignGUI gui;
-    private JTable personnelTable;
-    private PersonnelTableModel personnelModel;
-    private ResourceBundle resourceMap;
+    private final CampaignGUI gui;
+    private final JTable personnelTable;
+    private final PersonnelTableModel personnelModel;
+    private final ResourceBundle resourceMap;
 
-    public PersonnelTableMouseAdapter(CampaignGUI gui, JTable personnelTable,
+    protected PersonnelTableMouseAdapter(CampaignGUI gui, JTable personnelTable,
                                       PersonnelTableModel personnelModel) {
-        super();
         this.gui = gui;
         this.personnelTable = personnelTable;
         this.personnelModel = personnelModel;
         resourceMap = ResourceBundle.getBundle("mekhq.resources.PersonnelTableMouseAdapter", new EncodeControl()); //$NON-NLS-1$
     }
 
+    public static void connect(CampaignGUI gui, JTable personnelTable,
+            PersonnelTableModel personnelModel, JSplitPane splitPersonnel) {
+        new PersonnelTableMouseAdapter(gui, personnelTable, personnelModel) {
+            @Override
+            public void mouseClicked(MouseEvent e) {
+                if ((e.getButton() == MouseEvent.BUTTON1) && (e.getClickCount() == 2)) {
+                    int width = splitPersonnel.getSize().width;
+                    int location = splitPersonnel.getDividerLocation();
+                    int size = splitPersonnel.getDividerSize();
+                    if ((width - location + size) < PersonnelTab.PERSONNEL_VIEW_WIDTH) {
+                        // expand
+                        splitPersonnel.resetToPreferredSizes();
+                    } else {
+                        // collapse
+                        splitPersonnel.setDividerLocation(1.0);
+                    }
+                }
+            }
```

Comment:
Why are we overriding this here when it is only ever used here? Why not have it as part of the base class?

Reply:
This is so I do not have to include the splitter as a variable in the class.